Source/WebKit/UIProcess/glib/SystemSettingsManager.cpp

```diff
@@ -0,0 +1,165 @@
+{
+    return emptyString();
+}
+
+String SystemSettingsManager::xftRGBA() const
+{
+    return "rgb"_s;
+}
+
+int SystemSettingsManager::xftDPI() const
+{
+    return -1;
+}
+
+bool SystemSettingsManager::cursorBlink() const
+{
+    return true;
+}
+
+int SystemSettingsManager::cursorBlinkTime() const
+{
+    return 1200;
+}
+
+bool SystemSettingsManager::primaryButtonWarpsSlider() const
+{
+    return true;
+}
+
+bool SystemSettingsManager::overlayScrolling() const
+{
+    return true;
+}
+
+bool SystemSettingsManager::enableAnimations() const
+{
+    return true;
+}
+
+#endif // !PLATFORM(GTK)
+
+void SystemSettingsManager::initialize()
+{
+    static NeverDestroyed<SystemSettingsManager> manager;
+}
+
+void SystemSettingsManager::settingsDidChange()
+{
+    auto& oldState = SystemSettings::singleton().settingsState();
+    SystemSettings::State changedState;
+
+    auto themeName = this->themeName();
+    if (oldState.themeName != themeName)
+        changedState.themeName = themeName;
+
+    auto fontName = this->fontName();
+    if (oldState.fontName != fontName)
+        changedState.fontName = fontName;
+
+    auto xftAntialias = this->xftAntialias();
+    if (xftAntialias != -1 && oldState.xftAntialias != xftAntialias)
+        changedState.xftAntialias = xftAntialias;
+
+    auto xftHinting = this->xftHinting();
+    if (xftHinting != -1 && oldState.xftHinting != xftHinting)
+        changedState.xftHinting = xftHinting;
+
+    auto xftDPI = this->xftDPI();
+    if (xftDPI != -1 && oldState.xftDPI != xftDPI)
+        changedState.xftDPI = xftDPI;
+
+    auto xftHintStyle = this->xftHintStyle();
+    if (oldState.xftHintStyle != xftHintStyle)
+        changedState.xftHintStyle = xftHintStyle;
+
+    auto xftRGBA = this->xftRGBA();
+    if (oldState.xftRGBA != xftRGBA)
+        changedState.xftRGBA = xftRGBA;
+
+    auto cursorBlink = this->cursorBlink();
+    if (oldState.cursorBlink != cursorBlink)
+        changedState.cursorBlink = cursorBlink;
+
+    auto cursorBlinkTime = this->cursorBlinkTime();
+    if (oldState.cursorBlinkTime != cursorBlinkTime)
+        changedState.cursorBlinkTime = cursorBlinkTime;
+
+    auto primaryButtonWarpsSlider = this->primaryButtonWarpsSlider();
+    if (oldState.primaryButtonWarpsSlider != primaryButtonWarpsSlider)
+        changedState.primaryButtonWarpsSlider = primaryButtonWarpsSlider;
+
+    auto overlayScrolling = this->overlayScrolling();
+    if (oldState.overlayScrolling != overlayScrolling)
+        changedState.overlayScrolling = overlayScrolling;
+
+    auto enableAnimations = this->enableAnimations();
+    if (oldState.enableAnimations != enableAnimations)
+        changedState.enableAnimations = enableAnimations;
+
+    for (auto& processPool : WebProcessPool::allProcessPools())
+        processPool->sendToAllProcesses(Messages::SystemSettingsProxy::DidChange(changedState));
+
+    // SystemSettings::singleton().applySettings(WTFMove(changedState));
```

Comment:
Is this actually a wip patch or draft?

Reply:
I just left in the part I had an open discussion over.